### PR TITLE
Fix floodlights not turning off when out of power, and fixes sling veil not working on them

### LINKED ISSUE #12053: Flood lights dont depower once turned on and cut from grid
### Severity

Round Breaking

### Short Description

FLood Lights dont depower once cut off from the power grid if they are turned on and must be manually flicked off

### Reproduction

SLings thrawl me
people are building floodlamps
find one left alone and powered
cut the cables around it connecting it to the grid
cut the apc wires too to be sure
flood lamp still active and emitting light
cry becuase all slings are dead now

### Round ID

Round #36679

### Test Merges

#12016 #11946 #11955 #11768 #11425 #12014 #11864 #11879 #11999 #12022 #11708 #11973 #12050

### BYOND Username

ethan4303

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by adamsong**
# Github documenting your Pull Request

Fixes floodlights not turning off when power to them is cut
Fixes sling veil not working on floodlights

Fixes #12053 

# Changelog

:cl:  
bugfix: fixed floodlights not turning off
bugfix: fixed sling veil not working on floodlights
/:cl:

### CODE REVIEW

**Comment by Cartlord** [comment]
I was under the impression the SLing thing was intentional, since the wiki actually advises setting floodlights everywhere in the event of confirmed SLings.

**Comment by JamieD1** [comment]
> I was under the impression the SLing thing was intentional, since the wiki actually advises setting floodlights everywhere in the event of confirmed SLings.

Pretty sure floodlights are not supposed to turn off if they are hard wired.

**Review by JamieD1 — CHANGES_REQUESTED** [review]
Please make it that if a floodlight is hardwired and the ling uses the ability it sets it to its lowest range instead, maybe also make it flash.

File: yogstation/code/modules/antagonists/shadowling/shadowling_abilities.dm [hunk 1]
```diff
@@ -170,6 +170,10 @@
 			var/obj/mecha/M = LO
 			M.set_light(0)
 			M.lights = FALSE
+		if(istype(LO, /obj/machinery/power/floodlight))
+			var/obj/machinery/power/floodlight/FL = LO
+			FL.change_setting(1) // Turn off the floodlight
```
Comment (adamsong) [inline yogstation/code/modules/antagonists/shadowling/shadowling_abilities.dm]:
```suggestion
			FL.change_setting(2) // Set floodlight to lowest setting
```

**Review by SomeguyManperson — APPROVED** [review]
considering the current meta of putting floodlights under glass which is completely unaddressable by shadowlings having floodlights set to the lowest range (which is still quite large) should be enough
also floodlight code is old and jank and old

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**